src/main/clojure/clara/rules/compiler.clj

```diff
@@ -235,15 +235,21 @@
   "Compiles a sequence of constraints into a structure that can be evaluated.
 
    Callers may also pass a collection of equality-only-variables, which instructs
-   this function to only do an equality check on them rather than create a unification binding."
+   this function to only do an equality check on them rather than create a unification binding.
+
+   The returned expression evaluates to a two item vector, [bindings error], which contains
+   the evaluated bindings in the first argument and a map with information on any exception
+   that might have been thrown by evaluating a constraint as the second vector entry."
```

Comment:
I gave this some thought and I think I see a way out here.

Steps:
1. We can just eliminate the try-catch in build-constraints if we're compiling ClojureScript.
In essence, this would bring us back to the previous behavior in ClojureScript while keeping the improvement in Clojure.
2. We could then add an option to add such try-catch blocks for ClojureScript that would primarily be used at development time (and would create unreachable code warnings in some cases).
3. We can then add information to the "else" case where there the exception isn't from a constraint
in the engine.  The error message could specify that the exception might be from either Clara itself or user code if that is desired. If the session is a ClojureScript one the message could reference the option created in (2) above.  

Normally I'd be pretty reluctant to add the complication of another user-facing option, but this one would have the nice property that anyone who needs it can have it very directly pointed out by the exception message, and the user can then recompile with that option on if they want more information.  For immediate purposes, I'd be fine with just implementing (1) for now and logging work to do (2) and (3).

Thoughts?  @rbrush @carlosphillips